roadmap/implementers-guide/src/node/approval/approval-voting.md

```diff
@@ -0,0 +1,234 @@
+# Approval Voting
+
+## Protocol
+
+Input:
+  - `ApprovalVotingMessage::CheckAndImportAssignment`
+  - `ApprovalVotingMessage::CheckAndImportApproval`
+  - `ApprovalVotingMessage::ApprovedAncestor`
+
+Output:
+  - `ApprovalNetworkingMessage::DistributeAssignment`
+  - `ApprovalNetworkingMessage::DistributeApproval`
+  - `RuntimeApiMessage::Request`
+  - `ChainApiMessage`
+  - `AvailabilityRecoveryMessage::Recover`
+  - `CandidateExecutionMessage::ValidateFromExhaustive`
+
+## Functionality
+
+The approval voting subsystem is responsible for casting votes and determining approval of candidates and as a result, blocks.
+
+This subsystem wraps a database which is used to store metadata about unfinalized blocks and the candidates within them. Candidates may appear in multiple blocks, and assignment criteria are chosen differently based on the hash of the block they appear in.
+
+## Database Schema
+
+The database schema is designed with the following goals in mind:
+  1. To provide an easy index from unfinalized blocks to candidates
+  1. To provide a lookup from candidate hash to approval status
+  1. To be easy to clear on start-up. What has happened while we were offline is unimportant.
+  1. To be fast to clear entries outdated by finality
+
+Structs:
+
+```rust
+struct TrancheEntry {
+    tranche: DelayTranche,
```

Comment:
The definition of delay tranche should be added here, rather than the disputes module:

> Assignment proceeds in loosely timed rounds called DelayTranches roughly 12 times faster than block production, in which validators send assignment notices until all candidates have enough checkers assigned. Assignment tracks when approval votes arrive too and assigns more checkers if some checkers run late.